src/intrinsics/react-mocks/mocks.js

```diff
@@ -11,14 +11,60 @@
 
 import type { Realm } from "../../realm.js";
 import { parseExpression } from "babylon";
-import { ObjectValue, ECMAScriptFunctionValue } from "../../values/index.js";
+import { ObjectValue, ECMAScriptFunctionValue, ECMAScriptSourceFunctionValue } from "../../values/index.js";
 import { Get } from "../../methods/index.js";
 import { Environment } from "../../singletons.js";
+import { getReactElementSymbol } from "../../react/utils.js";
 import invariant from "../../invariant";
 
+// most of the code here was taken from https://github.com/facebook/react/blob/master/packages/react/src/ReactElement.js
 let reactCode = `
-  {
-    Component: class Component {
+  function createReact(REACT_ELEMENT_TYPE, ReactCurrentOwner) {
+    var hasOwnProperty = Object.prototype.hasOwnProperty;
+    var RESERVED_PROPS = {
+      key: true,
+      ref: true,
+      __self: true,
+      __source: true,
+    };
+
+    var ReactElement = function(type, key, ref, self, source, owner, props) {
+      return {
+        // This tag allow us to uniquely identify this as a React Element
+        $$typeof: REACT_ELEMENT_TYPE,
+    
+        // Built-in properties that belong on the element
+        type: type,
+        key: key,
+        ref: ref,
+        props: props,
+    
+        // Record the component responsible for creating this element.
+        _owner: owner,
+      };
+    };
+
+    function hasValidRef(config) {
+      return config.ref !== undefined;
+    }
+    
+    function hasValidKey(config) {
+      return config.key !== undefined;
+    }
+
+    class Component {
```

Comment:
This is a pretty simplified stub implementation of Component. What consequence does that have? Could it mean Prepack breaks some code it wouldn't otherwise? (I see it's not changed from before though.)

Reply:
This is only meant to be a mock/stub. The real logic get's handled by the Prepack React reconciliation code. The constructor of any of the components never gets called in fact.

Reply:
Can we make it throw then?

Reply:
Agree if it never should get called it should throw.

Reply:
Agreed. Done :)